pkg/usagetracker/tracker_store.go

```diff
@@ -135,7 +136,13 @@ func (t *trackerStore) getOrCreateTenantShard(userID string, shard uint8, limit
 		return m
 	}
 
-	m = &tenantShard{series: make(map[uint64]*atomic.Int32, int(limit/shards))}
+	var series map[uint64]*atomic.Int32
+	if limit == noLimit {
+		series = make(map[uint64]*atomic.Int32)
```

Comment:
Instead of not preallocating at all, should we use a large number? If a tenant has no limit, probably its number of series is pretty large.